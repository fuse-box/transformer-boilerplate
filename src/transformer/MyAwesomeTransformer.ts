import { ISchema } from "fuse-box/compiler/core/nodeSchema";
import { ASTType } from "fuse-box/compiler/interfaces/AST";
import { ITransformer } from "fuse-box/compiler/interfaces/ITransformer";

export default function MyAwesomeTransformer(myOptions): ITransformer {
  return {
    commonVisitors: props => ({
      onEach: (schema: ISchema) => {
        const { node, replace } = schema;
        if (node.type === ASTType.Literal) {
          if (node.value === "hello") {
            return replace({ type: "Literal", value: "world" });
          }
        }
      }
    })
  };
}
