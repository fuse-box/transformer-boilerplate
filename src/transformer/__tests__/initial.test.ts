import { testTransform } from "fuse-box";
import MyAwesomeTransformer from "../MyAwesomeTransformer";

describe("Awesome transformer test", () => {
  const test = (props: { code: string; jsx?: boolean }) => {
    return testTransform({
      code: props.code,
      jsx: props.jsx,
      transformers: [MyAwesomeTransformer({})]
    });
  };

  it("should", () => {
    const res = test({ code: `console.log("hello")` });
    expect(res.code).toMatchSnapshot()
  });
});
