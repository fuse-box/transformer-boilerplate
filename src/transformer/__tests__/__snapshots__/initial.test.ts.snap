// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Awesome transformer test should 1`] = `
"console.log(\\"world\\");
"
`;
